File: C/exo2.c
#include <stdio.h>

void swap(int *a, int *b);

int main(void)
{
    // on récupère le premier nombre
    int x = printf("Nombre 1 : ");
    scanf("%i", &x);

    // on récupère le second nombre
    int y = printf("Nombre 2 : ");
    scanf("%i", &y);

    // On affiche le résultat
    printf("Le nombre 1 vaut %i et le nombre 2 vaut %i\n", x, y);
    swap(&x, &y); // on passe l'adresse de x et celle de y
    printf("Le nombre 1 vaut %i et le nombre 2 vaut %i\n", x, y);
}

void swap(int *a, int *b)
{
    int tmp = *a;
    *a = *b;
    *b = tmp;
}

File: Python/exo2.py
# on demande le nombre 1
x = int(input("Nombre 1 : "))
# on demande le nombre 2
y = int(input("Nombre 2 : "))
# le résultat au début
print(f"Le nombre 1 vaut {x} et le nombre 2 vaut {y}")
# le calcul
x, y = y, x
# le résultat à la fin
print(f"Le nombre 1 vaut {x} et le nombre 2 vaut {y}")